airflow/models/xcom.py

```diff
@@ -108,6 +111,7 @@ def set(
         dag_id: str,
         task_id: str,
         run_id: str,
+        map_index: int = -1,
```

Comment:
I wonder if instead of all these columns we should take a `Union[TaskInstance, TaskInstanceKey]`?

Reply:
I was wondering about just `TaskInstanceKey`; it’s quite trivial to build a key from a ti anyway.

Reply:
It's `ti.key` infact :)

Reply:
Hmmm, I've started making this change and I'm not too sure about it. Will continue for now.